### PR TITLE
Reconcile is no longer triggered if there are no publishers in the ledger table.

### LINKED ISSUE #13325: Reconcile shouldn't happen when ledger table is empty
### Description
Reconcile shouldn't happen when ledger table is empty 

### Steps to Reproduce
  1. Clean install 0.21.16, **do not visit any site**
  2. Enable payment and add funds to wallet
  3. Change reconcile stamp to older date, wait for the reconcile to process
  4. Amount from wallet is deducted based on the monthly budget even with out any entry for ledger table

**Actual result:**
Reconcile happens even when ledger table is empty 

**Expected result:**
Reconcile shouldn't happen when ledger table is empty 

**Reproduces how often:**
100%

### Brave Version

**about:brave info:**

Brave | 0.21.16
-- | --
V8 | 6.4.388.41
rev | a7bed47
Muon | 4.8.2
OS Release | 10.0.16299
Update Channel | Beta
OS Architecture | x64
OS Platform | Microsoft Windows
Node.js | 7.9.0
Brave Sync | v1.4.2
libchromiumcontent | 64.0.3282.140

**Reproducible on current live release:**
Yes

### PR DESCRIPTION
**Pull request by ryanml**
Fixes #13325

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:
  1. Clean install 0.21.16, do not visit any site
  2. Enable payment and add funds to wallet
  3. Change reconcile stamp to older date.
  4. Confirm that the reconciliation does not process

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [x] Adequate test coverage exists to prevent regressions
- [x] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by ryanml** [comment]
@srirambv @NejcZdovc

**Comment by ryanml** [comment]
@NejcZdovc let me know if this is a change that makes more sense to implement in to the bat-client itself, and not browser-side. Thanks!

**Comment by NejcZdovc** [comment]
@ryanml well I think that solution is sadly not that simple. We need to define what actually happens in this case with contribution time should we push it back for 30 days or just fuzz it for couple of days. I think that if your table is empty I would push it back for 30 days. So we need to update reconcile timestamp.

**Comment by ryanml** [comment]
@NejcZdovc makes complete sense.

Would we just be updating `delayTime` in this case or setting `setTimeUntilReconcile` via the client? 

Either way, I'll work something out and we can review from there :)

**Comment by NejcZdovc** [comment]
@ryanml we should set `setTimeUntilReconcile`

**Comment by ryanml** [comment]
@NejcZdovc I think we're a little closer to the desired result now

File: app/browser/api/ledger.js [hunk 1]
```diff
@@ -2243,9 +2243,22 @@ const run = (state, delayTime) => {
     return
   }
 
+  const publishers = ledgerState.getPublisher(state)
+  if (publishers.isEmpty()) {
+    const newReconcileTime = (new Date().getTime() + (ledgerUtil.milliseconds.day * 30))
+    client.setTimeUntilReconcile(newReconcileTime, (err, stateResult) => {
```
Comment (NejcZdovc) [inline app/browser/api/ledger.js]: Let's extract this block into separate function and just call this function here. Also please replace this block in other places that you can find it with this new function. You don't need to pass newReconcileTime, you can just pass `null ` in, which will default to 30 days.

**Comment by ryanml** [comment]
@NejcZdovc changes pushed

File: app/browser/api/ledger.js [hunk 2]
```diff
@@ -2243,6 +2247,11 @@ const run = (state, delayTime) => {
     return
   }
 
+  const publishers = ledgerState.getPublisher(state)
+  if (publishers.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (ryanml) [reply app/browser/api/ledger.js]: @NejcZdovc good call, changes pushed

**Comment by NejcZdovc** [comment]
@ryanml let's just squash this and we are good to go

**Comment by ryanml** [comment]
@NejcZdovc squashed 👍 thanks for your guidance on this one!

**Comment by NejcZdovc** [comment]
@ryanml sure, not a problem

**Comment by NejcZdovc** [comment]
master bc3262a
0.23 372c933d1
0.22 331000eb8

**Pull request merged**